syntax = "proto3";

import "userdoc.proto";

message QueryError {
    ErrorType error_type = 1;
    string detail = 2;

    enum ErrorType {
        UNSPECIFIED = 0;
        NO_SUCH_USER = 1;
        INVALID_ID = 2;
      }
  }

message PutError {
    ErrorType error_type = 1;
    string detail = 2;

    enum ErrorType {
        UNSPECIFIED = 0;
        SCHEMA_FAIL = 1;
        ALREADY_EXISTS = 2;
      }
  }


message GetUser {
    oneof result {
        UserDocument user = 1;
        QueryError err = 2;
      }
  }

message CreatedUser {
    oneof result {
        UserDocument new_user = 1;
        PutError err = 2;
      }
  }
